python/semantic_kernel/connectors/ai/function_calling_utils.py

```diff
@@ -0,0 +1,189 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import logging
+from copy import copy
+from typing import TYPE_CHECKING, Any
+
+from semantic_kernel.connectors.ai.function_choice_behavior import (
+    FunctionChoiceBehavior,
+)
+from semantic_kernel.connectors.ai.open_ai.prompt_execution_settings.open_ai_prompt_execution_settings import (
+    OpenAIChatPromptExecutionSettings,
+)
+from semantic_kernel.contents.chat_history import ChatHistory
+from semantic_kernel.contents.function_call_content import FunctionCallContent
+from semantic_kernel.contents.function_result_content import FunctionResultContent
+from semantic_kernel.exceptions import (
+    FunctionCallInvalidArgumentsException,
+    FunctionExecutionException,
+)
+from semantic_kernel.filters.auto_function_invocation.auto_function_invocation_context import (
+    AutoFunctionInvocationContext,
+)
+from semantic_kernel.filters.filter_types import FilterTypes
+from semantic_kernel.filters.kernel_filters_extension import _rebuild_auto_function_invocation_context
+from semantic_kernel.functions.function_result import FunctionResult
+from semantic_kernel.functions.kernel_function_metadata import KernelFunctionMetadata
+
+if TYPE_CHECKING:
+    from semantic_kernel.connectors.ai.function_choice_behavior import (
+        FunctionCallChoiceConfiguration,
+    )
+    from semantic_kernel.connectors.ai.open_ai.prompt_execution_settings.open_ai_prompt_execution_settings import (
+        OpenAIChatPromptExecutionSettings,
+    )
+    from semantic_kernel.functions.kernel_arguments import KernelArguments
+    from semantic_kernel.kernel import Kernel
+
+logger = logging.getLogger(__name__)
+
+
+def update_settings_from_function_call_configuration(
+    function_choice_configuration: "FunctionCallChoiceConfiguration",
+    settings: "OpenAIChatPromptExecutionSettings",
+    type: str,
+) -> None:
+    """Update the settings from a FunctionChoiceConfiguration."""
+    if function_choice_configuration.available_functions:
+        settings.tool_choice = type
+        settings.tools = [
+            kernel_function_metadata_to_function_call_format(f)
+            for f in function_choice_configuration.available_functions
+        ]
+
+
+def kernel_function_metadata_to_function_call_format(
+    metadata: KernelFunctionMetadata,
+) -> dict[str, Any]:
+    """Convert the kernel function metadata to function calling format."""
+    return {
+        "type": "function",
+        "function": {
+            "name": metadata.fully_qualified_name,
+            "description": metadata.description or "",
+            "parameters": {
+                "type": "object",
+                "properties": {param.name: param.schema_data for param in metadata.parameters if param.is_required},
+                "required": [p.name for p in metadata.parameters if p.is_required],
+            },
+        },
+    }
+
+
+async def process_function_call(
```

Comment:
why does this not live in the Kernel?

Reply:
Moved it to the kernel with the name `invoke_function_call`.